mountpoint-s3/src/cli.rs

```diff
@@ -0,0 +1,250 @@
+use crate::build_info;
+
+use anyhow::anyhow;
+use clap::{CommandFactory, FromArgMatches, Parser};
+use mountpoint_s3_fs::cli::{CliArgs, CliArgsBase, MOUNT_OPTIONS_HEADER};
+use std::env;
+use std::ffi::OsString;
+
+// TODO: Extract `mountpoint_s3_fs::cli::CliArgs` to this crate.
+// TODO: Stabilise the library API
+#[derive(Parser, Debug)]
+#[clap(
+    name = "mount-s3",
+    about = "Mountpoint for Amazon S3",
+    version = build_info::FULL_VERSION,
+)]
+pub struct AppCliArgs {
+    #[clap(flatten)]
+    pub cli_args: CliArgs,
+}
+
+#[derive(Parser, Debug)]
+struct FsTabCliArgs {
+    #[clap(help = "Name of bucket to mount")]
+    pub bucket_name: String,
+
+    #[clap(
+        help = "Directory or FUSE file descriptor to mount the bucket at",
+        value_name = "DIRECTORY"
+    )]
+    pub mount_point: String,
+
+    #[clap(help = "FsTab style options", short = 'o')]
+    pub options: String,
+}
+
+impl FsTabCliArgs {
+    pub fn try_to_cli_args(self) -> anyhow::Result<CliArgs> {
+        let cli_arg_list = self.build_cli_arg_list()?;
+        Self::build_cli_args(cli_arg_list)
+    }
+
+    fn build_cli_arg_list(&self) -> anyhow::Result<impl Iterator<Item = OsString>> {
+        let options = Self::split_commas(&self.options)?
+            .into_iter()
+            .filter(|option| !Self::filter_option(option))
+            .map(|option| {
+                if option.starts_with("-") {
+                    option
+                } else {
+                    format!("--{}", option)
+                }
+            });
+        let cli_arg_list = [
+            env::args_os().nth(0).unwrap_or("mount-s3".into()),
```

Comment:
Could we avoid the binary name? See https://docs.rs/clap/latest/clap/struct.Command.html#method.no_binary_name